src/rabbit_env.erl

```diff
@@ -1490,18 +1494,54 @@ do_load_conf_env_file(Context, Sh, ConfEnvFile) ->
               use_stdio,
               stderr_to_stdout,
               exit_status]),
-    collect_sh_output(Context, Port, Marker, <<>>).
+    collect_sh_output(Context, Port, Marker, <<>>);
+do_load_conf_env_file(win32, Context, Cmd, ConfEnvFile) ->
+    rabbit_log_prelaunch:debug(
+      "Sourcing $RABBITMQ_CONF_ENV_FILE: ~ts", [ConfEnvFile]),
+
+    #{rabbitmq_base := RabbitmqBase,
+      rabbitmq_home := RabbitmqHome} = Context,
+    MainConfigFile = re:replace(
+                       get_default_main_config_file(Context),
+                       "\\.(conf|config)$", "", [{return, list}]),
+    Env = [
+           {"RABBITMQ_BASE", RabbitmqBase},
+           {"RABBITMQ_HOME", RabbitmqHome},
+           {"CONFIG_FILE", MainConfigFile},
+           {"ADVANCED_CONFIG_FILE", get_default_advanced_config_file(Context)},
+           {"MNESIA_BASE", get_default_mnesia_base_dir(Context)},
+           {"ENABLED_PLUGINS_FILE", get_default_enabled_plugins_file(Context)},
+           {"PLUGINS_DIR", get_default_plugins_path_from_env(Context)},
+           {"CONF_ENV_FILE_PHASE", "rabbtimq-prelaunch"}
+          ],
+
+    % rabbitmq/rabbitmq-common#392
+    % There is no way to double-quote the file path to deal with spaces because
+    % spawn_executable re-escapes them, resulting in the following:
+    % '\"C:\Users\lbakken\issues\misc space\rabbitmq-env-conf.bat\"' is not recognized as an internal or external command
+    % For now, assume users will not use paths with spaces.
+    Arg = io_lib:format("echo off && call ~ts && set", [ConfEnvFile]),
```

Comment:
Do you think we could use PowerShell instead, to solve escaping issues?

Reply:
I will try that. Great idea.

Reply:
Actually I figured out the syntax to work with spaces - https://github.com/rabbitmq/rabbitmq-common/pull/393/commits/79b605ffbf2c80924f79b019fa194af6966b7364